addons/im_livechat/models/im_livechat_channel.py

```diff
@@ -262,10 +262,15 @@ def _get_random_operator(self, lang=None, country_id=None):
         active_channels = self.env.cr.dictfetchall()
         operator = None
         # Try to match an operator with the same lang as the visitor
+        # If no operator with the same lang, try to match an operator with the preferred lang
         if lang:
             same_lang_operator_ids = self.available_operator_ids.filtered(lambda operator: operator.partner_id.lang == lang)
             if same_lang_operator_ids:
                 operator = self._get_less_active_operator(active_channels, same_lang_operator_ids)
+            else:
+                preferred_lang_operator_ids = self.available_operator_ids.filtered(lambda operator: lang in operator.livechat_langs.mapped('code'))
+                if preferred_lang_operator_ids:
+                    operator = self._get_less_active_operator(active_channels, preferred_lang_operator_ids)
```

Comment:
this is exactly the same line as before, maybe it could be factorised not to repeat it?

Reply:
I don't think this is too much duplicated because it's assigning value with an already refactored method. There is a lot of other places calling the same method as well and that's why we are using `_get_less_active_operator`